engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/EqualsMultiValueTest.java

```diff
@@ -79,9 +79,13 @@ public void testEqualityIntegerAndTextMultiValue() {
               EnsoMultiValue.create(
                   new Type[] {textText, intType}, 1, new Object[] {Text.create("Hi"), 4L});
 
-          assertFalse("4 != t4", equalityCheck(4L, fourExtraText));
+          assertTrue(
+              "Only Text is 'dispatch type'. Not integer. Shall we be equal to 4?",
+              equalityCheck(4L, fourExtraText));
```

Comment:
**Open question**: what's the desired behavior in this `(x : Text & Integer) : Text` case? 
- `assertFalse("4 != t4", equalityCheck(4L, fourExtraText));` 
- `assertTrue("4 == t4",equalityCheck(4L, fourExtraText));`

Does the final object can still be equal - e.g. represent `Integer` (without being casted to it) - or it can only be equal to `Text`?

Reply:
Do we still stand with `obj : Integer & Text` to be equal to both Text and Integer in light of [this comment of yours](https://github.com/enso-org/enso/pull/11600#discussion_r1858570367)?

I was also unsure, how does that play with our "Enso" `==`? If I do `obj == x`, do I then actually end up calling `((obj : Text) == x) || ((obj : Integer) == x)`?